### PR TITLE
Add Model Tracker and Updater along with supporting interface refactors.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**

### CODE REVIEW

File: .vscode/settings.json
Comment (RusticLuftig) [inline .vscode/settings.json:1]: This shouldn't be check in. Please double check why the .gitignore isn't catching this

File: model/model_tracker.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def get_model_id_for_miner_hotkey(self, hotkey: str) -> Optional[ModelId]:
+        """Returns the model id for a given hotkey if any."""
+
+        with self.lock:
+            if hotkey in self.miner_hotkey_to_model_id_dict:
+                return self.miner_hotkey_to_model_id_dict[hotkey]
+            return None
+
+    def on_hotkeys_updated(self, incoming_hotkeys: Set[str]):
+        """Notified the tracker which hotkeys are currently being tracked on the metagraph."""
```
Comment (RusticLuftig) [inline model/model_tracker.py:56]: uber nit: Notifies

File: model/model_tracker.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        with self.lock:
+            existing_hotkeys = set(self.miner_hotkey_to_model_id_dict.keys())
+            for hotkey in existing_hotkeys - incoming_hotkeys:
+                del self.miner_hotkey_to_model_id_dict[hotkey]
```
Comment (RusticLuftig) [inline model/model_tracker.py:61]: WDYT about adding a trace log here?

File: model/storage/utils.py [hunk 3]
```diff
@@ -2,10 +2,11 @@
 from model.data import ModelId
 
 
-# TODO make this configurable.
-def get_local_miner_dir(hotkey: str) -> str:
-    return os.path.join("local-models", hotkey)
+def get_local_miner_dir(base_dir: str, hotkey: str) -> str:
```
Comment (RusticLuftig) [inline model/storage/utils.py:5]: General question: Where do we ensure the relevant directories are created?

File: tests/model/storage/fake_model_metadata_store.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import os
+from model.data import ModelId, ModelMetadata
+from model.storage.chain import constants
+from model.storage.model_metadata_store import ModelMetadataStore
+from typing import Optional
+
+from utilities import utils
+
+
+class FakeModelMetadataStore(ModelMetadataStore):
```
Comment (RusticLuftig) [inline tests/model/storage/fake_model_metadata_store.py:13]: <3

**Review by RusticLuftig — CHANGES_REQUESTED** [review]
Overall LGTM. Just a few minor nits

File: .vscode/settings.json
Comment (Sid-Data-Universe) [reply .vscode/settings.json:1]: The .vscode/ ignore was added after. I actually noticed the merge conflict and added it in but didn't think to check why it had to get added in the first place.

File: model/storage/utils.py (hunk 3 above)
Comment (Sid-Data-Universe) [reply model/storage/utils.py:5]: They get created the first time a file is written. So if we don't have local-models and we write to local-models/hotkey0/namespace/model/files then it creates all of those as needed.

File: model/model_tracker.py (hunk 2 above)
Comment (Sid-Data-Universe) [reply model/model_tracker.py:61]: Done. Figure it will be rare so just added one after the delete.

**Review by RusticLuftig — APPROVED** [review]

**Pull request merged**